core_types/src/eav.rs

```diff
@@ -19,13 +19,34 @@ use std::{
 };
 
 use regex::RegexBuilder;
-use std::fmt::Debug;
+use std::fmt::{self, Debug};
 
 /// Address of AddressableContent representing the EAV entity
 pub type Entity = Address;
 
-/// Using String for EAV attributes (not e.g. an enum) keeps it simple and open
-pub type Attribute = String; // BUT we totally could make it an enum with a Custom variant!
+/// All Attribute values are pre-defined here. If ever a user-defined Attribute is needed,
+/// just add a new Custom variant for it with a String parameter
+#[derive(PartialEq, Eq, PartialOrd, Hash, Clone, Debug, Serialize, Deserialize, DefaultJson)]
+#[serde(rename_all = "lowercase")]
```

Comment:
Sorry, I actually meant:
```suggestion
#[serde(rename_all = "snake_case")]
```